rotkehlchen/exchanges/gemini.py

```diff
@@ -511,19 +513,18 @@ def query_online_deposits_withdrawals(
                 timestamp = deserialize_timestamp(entry['timestampms'])
                 timestamp = Timestamp(int(timestamp / 1000))
                 asset = asset_from_gemini(entry['currency'])
-
+                # Gemini does not include withdrawal fees neither in the API nor in their UI
                 movement = AssetMovement(
                     location=Location.GEMINI,
-                    category=deserialize_asset_movement_category(entry['type']),
-                    address=deserialize_asset_movement_address(entry, 'destination', asset),
-                    transaction_id=get_key_if_has_val(entry, 'txHash'),
-                    timestamp=timestamp,
+                    event_type=deserialize_asset_movement_event_type(entry['type']),
+                    timestamp=ts_sec_to_ms(timestamp),
                     asset=asset,
-                    amount=deserialize_asset_amount_force_positive(entry['amount']),
-                    fee_asset=asset,
-                    # Gemini does not include withdrawal fees neither in the API nor in their UI
-                    fee=Fee(ZERO),
-                    link=str(entry['eid']),
+                    balance=Balance(deserialize_asset_amount_force_positive(entry['amount'])),
+                    unique_id=str(entry['eid']),
+                    extra_data=maybe_set_transaction_extra_data(
+                        address=deserialize_asset_movement_address(entry, 'destination', asset),
+                        transaction_id=get_key_if_has_val(entry, 'txHash'),
```

Comment:
same question about combining the tx hash with `eid` in this case.

Reply:
same here, I think `eid` should be unique on its own. Also, since `txHash` is being retrieved with `get_key_if_has_val` it would seem that it may not always be present as well.